### ISSUE TITLE
s4ga FPGA

### ISSUE THREAD

**Issue by psychogenic** [TinyTapeout/tinytapeout-03p5#13]
I have tried to migrate the S4GA fpga to tt03p5, and the code is here:

https://github.com/psychogenic/tt03p5-s4ga

However, when I run tests locally they are failing and I _do_not_ trust/understand why I'm getting the green light here.

I don't _think_ I broke the thing in migrating it, but I'm uncertain, I don't quite get how this thing works.
@mattvenn : would like some input on this from source, or you, if you grok it.

```
**************************************************************************************
** TEST                          STATUS  SIM TIME (ns)  REAL TIME (s)  RATIO (ns/s) **
**************************************************************************************
** test.test_s4ga                 FAIL     3250000.00           0.04   72227074.54  **
**************************************************************************************
** TESTS=1 PASS=0 FAIL=1 SKIP=0            3250000.00           0.11   28633669.78  **
**************************************************************************************
```

**Comment by mattvenn** [comment]
original repo is here: https://github.com/grayresearch/tt02-s4ga (before it was ported to tt03 by someone else).
I cloned the original repo and tests work locally.

local tests also pass on the updated (tt03) version:
https://github.com/bat52/tt03-s4ga

I cloned Pat's repo and the test fails. So I would guess there is a real problem most likely related to the inverted reset.

**Comment by mattvenn** [comment]
I tried to leave a comment but it got lost somehow. I tried the original tests and they work, so I assume this is showing a real problem and it's most likely related to inverted reset.
Not worth taping out on tt3p5 unless the error is resolved.

**Comment by psychogenic** [comment]
Yeah, I suspected as much as well, and played with it a few ways.  In current version
 * it's passed as-is from testbench (tb.v) to dut (s4ga.v)
 * in dut rst_n is flipped to preserve original logic
but I have explored a whole lot of other approaches and combination, no avail.

My feeling is that this is actually somehow caused by the declaration
```
output reg  [7:0] uo_out
```
which I had to change from wire just to get it to stop griping about 
```
error: uo_out['sd7] is not a valid l-value in tb.s4ga.
```
and such, but I don't know thefull implications in verilog.
